feat(build): implement build command

BeniCheni opened the pull request:
Refer to [tink-build topic](https://npm.community/t/tink-implement-tink-build/3313).

- [x] Implementation
- [x] Add unit test

##### Local Testing

###### Without `build` run-script
No script is run.
```sh
→ tink build
```

###### With a dummy script, i.e. `"build": "echo Bingo!"`
`build` run-script is executed to print `Bingo!`.
```sh
→ tink build

> tink@0.14.0 build /Users/bchen/github/tink
> echo Bingo!

Bingo!
```

BeniCheni committed:
feat(build): implement build command

BeniCheni commented on bin/tink.js:
```diff
@@ -4,12 +4,13 @@ require('../lib/node/index.js')
   ['ping', require('../lib/commands/ping.js')],
-  ['deprecate', require('../lib/commands/deprecate.js')]
+  ['rm', require('../lib/commands/rm.js')],
+  ['shell', require('../lib/commands/shell.js')]
```
Minor FYI: ☝️while adding the required `build.js`, why not sort these items alphabetically.
(Happy to revert sort if it's bad idea.)

BeniCheni commented on lib/commands/build.js:
```diff
@@ -0,0 +1,56 @@
+    group: {},
+    stdio: {},
+    scriptShell: {}
+  })
```
Minor FYI: these default values of the `BuildConfig` are based on my best guesses, while reading code about how these option keys are being used in [npm-lifecycle/index.js](https://github.com/npm/npm-lifecycle/blob/latest/index.js).

Please advise if any of these default values, or the way of how this config object is written, can be improved. 🙏

zkat commented on lib/commands/build.js:
```diff
@@ -0,0 +1,56 @@
+}
+
+async function build (argv) {
+  const findPrefix = require('find-npm-prefix')
```
This is available as `libnpm.getPrefix()`

zkat commented on lib/commands/build.js:
```diff
@@ -0,0 +1,56 @@
+
+async function build (argv) {
+  const findPrefix = require('find-npm-prefix')
+  const readJson = require('read-package-json')
```
This one is available as `libnpm.readJSON()`

zkat commented on lib/commands/build.js:
```diff
@@ -0,0 +1,56 @@
+      false,
+      async (er, data) => {
+        if (er) {
+          console.error(`There was an error reading ${packageJson}`)
```
`libnpm.log.error()` for this and the one below, please.

zkat reviewed (changes requested):
Just a few small nits so far. Looking forward to having tests for it! 💚 Everything looks great and I appreciate the contribution!

BeniCheni committed:
refactor(build): use libnpm.* for utilities

BeniCheni committed:
refactor(build): update code to support writing unit test(s)

BeniCheni committed:
test(build): write a test to cover basic code coverage

BeniCheni commented:
Hi @zkat. Updated per the great nits you pointed out, updated the source to better support testability & wrote a basic test to assert `true` to validate the successful execution of the `build` command. Hopefully, this `build` PR is in good shape for review again for a wrap. Please review at convenience.

###### Trivial Observation
> File                |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s 
> build.js          |    93.33 |      100 |      100 |    93.33 |                 7 

The uncovered line 7 is
```js
return y.help().alias('help', 'h')
```
which are also observed for the `deprecate.js` & `ping.js` tests. So wondering if this uncovered line observation is rather "_universal_" among some tests. So I left it as it is for now, as % coverage threshold has not been set up yet.

BeniCheni committed:
fix(build): remove dummy "build" run-script

BeniCheni commented:
Ping/CC @zkat. Friendly FYI the branch of this PR has been rebased with `latest`, after the recent merge of other PRs. This PR should be in good shape for final review.

zkat commented on lib/commands/build.js:
```diff
@@ -0,0 +1,43 @@
+  return await libnpm.getPrefix(process.cwd())
+    .then(prefix => {
+      const packageJSON = `${prefix}/package.json`
+      libnpm.readJSON(packageJSON)
```
You should use an `await` here as well and get rid of the .then()s

BeniCheni committed:
refactor(build): adapt await to omit .then syntax

BeniCheni commented:
@zkat, indeed, I was able to improve the code to adapt `await` syntax to omit the `.then` usage in [this commit](https://github.com/npm/tink/pull/29/commits/94ec2a62bab2d8ffe6f3a409070e0309f222d75d). Please take a quick look at convenience. Thanks!

BeniCheni committed:
refactor(build): remove unneeded new line
